tests/unit/providers/test_aws_provider.py

```
@@ -130,6 +130,55 @@ def test_no_api_key_skips_bearer_token_setup(monkeypatch: pytest.MonkeyPatch) ->
+) -> None:
+    """Test that explicit AWS credential kwargs satisfy verification (regression test for #1183).
+
+    A bare `boto3.Session()` only resolves credentials from the ambient default chain (env vars,
+    `~/.aws/credentials`, instance role), so it must not be what verification relies on: it should
+    fail here since there are no ambient credentials, but verification must still succeed because
+    `aws_access_key_id`/`aws_secret_access_key`/`aws_session_token`/`region_name` were passed
+    explicitly to the constructor.
+    """
+    monkeypatch.delenv("AWS_BEARER_TOKEN_BEDROCK", raising=False)
+
+    def session_factory(**kwargs: Any) -> Mock:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)